runtime/avn/src/governance/tracks.rs

```diff
@@ -0,0 +1,142 @@
+        },
+    ),
+    (
+        1,
+        pallet_referenda::TrackInfo {
+            name: "whitelisted_caller",
+            max_deciding: 100,
+            decision_deposit: 1000 * AVT,
+            prepare_period: 30 * MINUTES,
+            decision_period: 28 * DAYS,
+            confirm_period: 10 * MINUTES,
+            min_enactment_period: 10 * MINUTES,
+            min_approval: APP_WHITELISTED_CALLER,
+            min_support: SUP_WHITELISTED_CALLER,
+        },
+    ),
+    (
+        2,
+        pallet_referenda::TrackInfo {
+            name: "general_admin",
+            max_deciding: 10,
+            decision_deposit: 500 * AVT,
+            prepare_period: 2 * HOURS,
+            decision_period: 28 * DAYS,
+            confirm_period: 3 * HOURS,
+            min_enactment_period: 10 * MINUTES,
+            min_approval: APP_GENERAL_ADMIN,
+            min_support: SUP_GENERAL_ADMIN,
+        },
+    ),
+    (
+        3,
+        pallet_referenda::TrackInfo {
+            name: "referendum_canceller",
+            max_deciding: 1000,
+            decision_deposit: 1000 * AVT,
+            prepare_period: 2 * HOURS,
+            decision_period: 7 * DAYS,
+            confirm_period: 3 * HOURS,
+            min_enactment_period: 10 * MINUTES,
+            min_approval: APP_REFERENDUM_CANCELLER,
+            min_support: SUP_REFERENDUM_CANCELLER,
+        },
+    ),
+    (
+        4,
+        pallet_referenda::TrackInfo {
+            name: "referendum_killer",
+            max_deciding: 1000,
+            decision_deposit: 2000 * AVT,
+            prepare_period: 2 * HOURS,
+            decision_period: 28 * DAYS,
+            confirm_period: 3 * HOURS,
+            min_enactment_period: 10 * MINUTES,
+            min_approval: APP_REFERENDUM_KILLER,
+            min_support: SUP_REFERENDUM_KILLER,
+        },
+    ),
+];
+
+pub struct TracksInfo;
+impl pallet_referenda::TracksInfo<Balance, BlockNumber> for TracksInfo {
+    type Id = u16;
+    type RuntimeOrigin = <RuntimeOrigin as frame_support::traits::OriginTrait>::PalletsOrigin;
+    fn tracks() -> &'static [(Self::Id, pallet_referenda::TrackInfo<Balance, BlockNumber>)] {
+        &TRACKS_DATA[..]
+    }
+    fn track_for(id: &Self::RuntimeOrigin) -> Result<Self::Id, ()> {
+        if let Ok(system_origin) = frame_system::RawOrigin::try_from(id.clone()) {
+            match system_origin {
+                frame_system::RawOrigin::Root => Ok(0),
+                _ => Err(()),
+            }
+        } else if let Ok(custom_origin) = origins::Origin::try_from(id.clone()) {
+            match custom_origin {
+                origins::Origin::WhitelistedCaller => Ok(1),
+                // General admin
+                origins::Origin::GeneralAdmin => Ok(2),
+                // Referendum admins
+                origins::Origin::ReferendumCanceller => Ok(3),
+                origins::Origin::ReferendumKiller => Ok(4),
+            }
+        } else {
```

Comment:
why do we need two matches here? One would suffice. It feels odd.
Is it to group the system and custom origins conceptually?

Reply:
yes it is